controllers/front/SearchController.php

```diff
@@ -227,4 +231,29 @@ public function setMedia()
             $this->addCSS(_THEME_CSS_DIR_.'product_list.css');
         }
     }
+    
+    /**
+     * findFirstClosestWord
+     * 
+     * @param $searchString
+     * 
+     * @return string
+     */
+    static function findFirstClosestWord($searchString)
+    {
+        $sql = 'SELECT `word` FROM `'._DB_PREFIX_.'search_word` 
+                WHERE `id_lang` = '.(int)Context::getContext()->language->id.'
+                AND `id_shop` = '.(int)Context::getContext()->shop->id.';';
+
+        $words = Db::getInstance()->executeS($sql);
+
+        foreach ( $words as &$item )
+            $item['leven'] = levenshtein($item['word'], $searchString);
+
+        $closestWord =  array_reduce($words, function($a, $b){
+            return $a['leven'] < $b['leven'] ? $a : $b;
+        }, array_shift($words))['word'];
+
+        return $closestWord;
+    }
```

Comment:
This method could be optimized. The two iterations over $words array are unnecessary. We could do this in one iteration simply by remembering word with lowest 'level'. This would also reduce memory complexity from O(n) to O(1).

Another suggestion is to query only words that have appropriate length (not necessarily same length, but it should be close enough). It doesn't make much sense to compare words 'thirtybees' with 'the', as the difference will always be huge.